subsys/net/lib/lwm2m/lwm2m_rw_cbor.c

```diff
@@ -0,0 +1,558 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define LOG_MODULE_NAME net_lwm2m_cbor
+#define LOG_LEVEL CONFIG_LWM2M_LOG_LEVEL
+
+#include <logging/log.h>
+#include <sys/util.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <time.h>
+LOG_MODULE_REGISTER(LOG_MODULE_NAME);
+
+#include "cbor_common.h"
+#include "cbor_decode.h"
+#include "cbor_encode.h"
+#include "lwm2m_object.h"
+#include "lwm2m_rw_cbor.h"
+#include "lwm2m_engine.h"
+#include "lwm2m_util.h"
+
+#define CPKT_CBOR_W_SZ(pos, cpkt) \
+	((size_t)(pos) - (size_t)(cpkt)->data - (size_t)(cpkt)->offset)
+
+enum  cbor_tag { /* www.iana.org/assignments/cbor-tags/cbor-tags.xhtml */
```

Comment:
Not found from cddl-gen

Reply:
@oyvindronningstad Worth of adding the to library itself?

Reply:
Good idea